### PR TITLE
chore(frontend): Integrate with tailwindcss. Fix #6264

### LINKED ISSUE #6264: Consider using tailwindcss

### PR DESCRIPTION
**Pull request by zijianjoy**
**Description of your changes:**

Fix #6264

Integration reference
https://tailwindcss.com/docs/guides/create-react-app
https://www.hoshki.me/blog/2020-09-11-how-to-setup-tailwind-css-with-cra-and-typescript/

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

File: frontend/.gitignore [hunk 1]
```diff
@@ -28,3 +28,6 @@ coverage
 
 # vscode
 .vscode
+
+# tailwindcss
+src/tailwind.output.css
```
Comment (Bobgy) [inline frontend/.gitignore:33]: Can we put it into a build or generated folder and ignore the folder instead?

Comment (zijianjoy) [reply frontend/.gitignore]: Sounds good, created build folder to put this file to there.

File: frontend/src/Css.tsx [hunk 2]
```diff
@@ -315,6 +315,10 @@ export const commonCss = stylesheet({
   },
 });
 
+export const tailwindcss = {
+  sideNavItem: 'flex flex-row flex-shrink-0',
```
Comment (Bobgy) [inline frontend/src/Css.tsx:319]: Do we need a common style file for non reusable styles? It seems to me that the tailwind css way is to directly write these classes in the component.

File: frontend/src/components/SideNav.tsx [hunk 3]
```diff
@@ -424,10 +432,12 @@ export class SideNav extends React.Component<SideNavInternalProps, SideNavState>
...
-                  Recurring Runs
-                </span>
+                <div className={tailwindcss.sideNavItem}>
+                  <Alarm />
+                  <span className={classes(collapsed && css.collapsedLabel, css.label)}>
+                    Recurring Runs
+                  </span>
+                </div>
               </Button>
             </Link>
```
Comment (Bobgy) [inline frontend/src/components/SideNav.tsx:442]: /lgtm

**Review by Bobgy — COMMENTED** [review]
/lgtm

File: frontend/src/Css.tsx (hunk 2 above)
Comment (zijianjoy) [reply frontend/src/Css.tsx]: It makes sense to move it to the same file where it uses.

Yes I struggled whether I should extract the tailwind class names or just define them in place. I eventually tend to create a const for being consistent with the rest of classNames approaches in the same file.

**Pull request merged**